mailbox/lucene/src/main/java/org/apache/james/mailbox/lucene/search/LuceneMessageSearchIndex.java

```diff
@@ -738,7 +758,8 @@ public void body(BodyDescriptor desc, InputStream in) throws IOException {
                     try (BufferedReader bodyReader = new BufferedReader(new InputStreamReader(in, charset))) {
                         String line = null;
                         while ((line = bodyReader.readLine()) != null) {
-                            doc.add(new TextField(BODY_FIELD, line.toUpperCase(Locale.US), Store.NO));
+                            System.out.println("index body: " + line);
+                            doc.add(new TextField(BODY_FIELD, line.toUpperCase(Locale.US), Store.YES));
```

Comment:
2 points concern:

1. Right now BODY_FIELD is not stored -> we cannot get body value when query
2. It is indexing line by line. 
Eg: 
When body:
```
             Summary: Convert Message into JSON
                 Key: MAILBOX-234
                 URL: https://issues.apache.org/jira/browse/MAILBOX-234
```
-> it is add 3 text BODY_FIELD here, 

For highlight fragment result, we need to index body single time

Reply:
> Right now BODY_FIELD is not stored -> we cannot get body value when query

 Let's store it 
 
 > it is add 3 text BODY_FIELD here,
 
 Looks inneficient. Can't we just have one field? We can remove reparators (ie `r` and `n` prior to sending this to lucene...

Reply:
Stupid question: does the Lucene indexer strip HTML tags prior to indexing?

Reply:
> Stupid question: does the Lucene indexer strip HTML tags prior to indexing?

My search result:
It depends on which analyzer we're using. 
Lucene provide class `HTMLStripCharFilter`